backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceMetrics.java

```diff
@@ -0,0 +1,309 @@
+     * Custom attributes provided by the user to BacktraceBase
+     */
+    protected Map<String, Object> customAttributes;
+
+    /**
+     * The application context
+     */
+    protected Context context;
+
+    /**
+     * Backtrace metrics settings
+     */
+    protected BacktraceMetricsSettings settings;
+
+    /**
+     * Create new Backtrace metrics instance
+     * @param context                   Application context
+     * @param customAttributes          Backtrace client
+     * @param settings
+     */
+    public BacktraceMetrics(Context context, Map<String, Object> customAttributes, BacktraceMetricsSettings settings) {
+        this.context = context;
+        this.customAttributes = customAttributes;
+        this.settings = settings;
+    }
+
+    @Override
+    public void startMetricsEventHandlers(Api backtraceApi) {
+        uniqueEventsHandler = backtraceApi.enableUniqueEvents(context, customAttributes, settings);
+        summedEventsHandler = backtraceApi.enableSummedEvents(context, customAttributes, settings);
+    }
+
+    void setStartupUniqueEventName(String StartupUniqueEventName) {
+        this.startupUniqueEventName = StartupUniqueEventName;
+    }
+
+    String getStartupUniqueEventName() {
+        return this.startupUniqueEventName;
+    }
+
+    void setMaximumEvents(int MaximumEvents) {
+        this.maximumEvents = MaximumEvents;
+    }
+
+    int getMaximumEvents() {
+        return this.maximumEvents;
+    }
+
+    public String getBaseUrl() {
+        return this.settings.getBaseUrl();
+    }
+
+    @SuppressWarnings("unchecked, rawtypes")
+    public ConcurrentLinkedDeque<UniqueEvent> getUniqueEvents() {
+        for (Event event : uniqueEventsHandler.events) {
+            if (!(event instanceof UniqueEvent)) {
+                BacktraceLogger.e(LOG_TAG, "UniqueEventsHandler contains an event of type " + event.getClass() + " this is a bug");
+                return new ConcurrentLinkedDeque<>();
+            }
+        }
+        return (ConcurrentLinkedDeque) this.uniqueEventsHandler.events;
+    }
+
+    @SuppressWarnings("unchecked, rawtypes")
+    public ConcurrentLinkedDeque<SummedEvent> getSummedEvents() {
+        for (Event event : summedEventsHandler.events) {
+            if (!(event instanceof SummedEvent)) {
+                BacktraceLogger.e(LOG_TAG, "SummedEventsHandler contains an event of type " + event.getClass() + " this is a bug");
+                return new ConcurrentLinkedDeque<>();
+            }
+        }
+        return (ConcurrentLinkedDeque) this.summedEventsHandler.events;
+    }
+
+    /**
+     * Send startup event to Backtrace
+     */
+    public void sendStartupEvent() {
+        uniqueEventsHandler.sendStartupEvent(startupUniqueEventName);
+        summedEventsHandler.sendStartupEvent(startupSummedEventName);
+    }
+
+    /**
+     * Send all outgoing messages (unique and summed) currently queued
+     */
+    public void send() {
+        uniqueEventsHandler.send();
```

Comment:
Should this do send async? Now it's concurrent.

Reply:
@vlussenburg @konraddysput this is async. `send` method adds a message to the `HandlerThread` queue which handles async sending. It's exactly the same behavior as sending the `BacktraceReport` now. See `BacktraceHandlerThread.java`

Reply:
Marking this resolved @vlussenburg @konraddysput